tests/handlers/aws/test_handler.py

```diff
@@ -127,104 +156,110 @@ def setUp(self) -> None:
-
-        assert first_call == "continuing"
-
-        while self._es_client.count(index="logs-redis.log-default")["count"] < 1:
-            pass
-
-        res = self._es_client.search(index="logs-redis.log-default")
-        assert res["hits"]["total"] == {"value": 1, "relation": "eq"}
-        assert (
-            res["hits"]["hits"][0]["_source"]["fields"]["message"]
-            == "79191:C 08 Jul 2021 13:25:02.609 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo"
-        )
-        assert res["hits"]["hits"][0]["_source"]["fields"]["log"] == {
-            "offset": 0,
-            "file": {"path": "https://test-bucket.s3.eu-central-1.amazonaws.com/redis.log.gz"},
-        }
-        assert res["hits"]["hits"][0]["_source"]["fields"]["aws"] == {
-            "s3": {
-                "bucket": {"name": "test-bucket", "arn": "arn:aws:s3:::test-bucket"},
-                "object": {"key": "redis.log.gz"},
-            }
-        }
-        assert res["hits"]["hits"][0]["_source"]["fields"]["cloud"] == {"provider": "aws", "region": "eu-central-1"}
-
-        event = self._event_from_sqs_message()
-        second_call = lambda_handler(event, ctx)  # type:ignore
-
-        assert second_call == "continuing"
-
-        while self._es_client.count(index="logs-redis.log-default")["count"] < 2:
-            pass
-
-        res = self._es_client.search(index="logs-redis.log-default")
-        assert res["hits"]["total"] == {"value": 2, "relation": "eq"}
-        assert (
-            res["hits"]["hits"][1]["_source"]["fields"]["message"]
-            == "79191:C 08 Jul 2021 13:25:02.610 # Redis version=6.2.4, bits=64, commit=00000000, "
-            + "modified=0, pid=79191, just started"
-        )
-
-        assert res["hits"]["hits"][1]["_source"]["fields"]["log"] == {
-            "offset": 81,
-            "file": {"path": "https://test-bucket.s3.eu-central-1.amazonaws.com/redis.log.gz"},
-        }
-        assert res["hits"]["hits"][1]["_source"]["fields"]["aws"] == {
-            "s3": {
-                "bucket": {"name": "test-bucket", "arn": "arn:aws:s3:::test-bucket"},
-                "object": {"key": "redis.log.gz"},
-            }
-        }
-        assert res["hits"]["hits"][1]["_source"]["fields"]["cloud"] == {"provider": "aws", "region": "eu-central-1"}
-
-        event = self._event_from_sqs_message()
-        third_call = lambda_handler(event, ctx)  # type:ignore
-
-        assert third_call == "completed"
+                            ),
+                            "eventSource": "aws:sqs",
+                            "eventSourceARN": self._source_queue_info["QueueArn"],
+                        },
+                    ]
+                }
+
+                first_call = lambda_handler(event, ctx)  # type:ignore
+
+                assert first_call == "continuing"
+
+                while self._es_client.count(index="logs-redis.log-default")["count"] < 1:
+                    pass
```

Comment:
How long does it usually take for this condition to be met? I was thinking to add a timeout here to prevent this test to hang up infinitely in case of issues and a sleep to go easier on the CPU.

Reply:
there's a `timeout 5m` in the script running the test in docker:
https://github.com/elastic/elastic-serverless-forwarder/blob/f440f1b05021bc36b3780777673a18ab8464e5d1/tests/scripts/docker/run_tests.sh#L22

I'd rather force the timeout externally when running the test that explicitly declare the timeout in the test, since during the development and debugging I had to tweak it.

but open to consider different solution

Reply:
this test takes less than 1m running directly, a little more than 2mins in CI.

the main waiting time is starting the docker containers

Reply:
`self._es_client.count(index="logs-redis.log-default")["count"] < 1` in particular is going to wait approximately for `index.refresh_interval` value (https://www.elastic.co/guide/en/elasticsearch/reference/7.15/index-modules.html#index-refresh-interval-setting)

Reply:
@endorama what do you think?

Reply:
I've force a `refresh` on the index, so I can now directly assert on the `count` api

Reply:
Current solutions seems perfect! Nice work.